### PR TITLE
Feature/max stakes

### PR DESCRIPTION
**Pull request by pingustar**

### CODE REVIEW

File: src/components/protection/StakeButtons.vue [hunk 1]
```diff
@@ -68,11 +108,37 @@ export default class StakeButtons extends Vue {
...
 
+  async loadMaxStakes() {
+    if (this.loadingMaxStakes || !this.poolId) return
+    this.loadingMaxStakes = true;
+    try {
+      const result = await vxm.ethBancor.getMaxStakes({
+        poolId: this.poolId
+      });
+      const maxStakes = result.maxStakesConverted
+      this.maxStakesBnt = prettifyNumber(maxStakes[Object.keys(maxStakes)[0]]);
```
Comment (Velua) [inline src/components/protection/StakeButtons.vue:125]: We cannot rely on key ordering for this to work, it's not always guaranteed.

File: src/store/modules/swap/ethBancor.ts [hunk 2]
```diff
@@ -3388,10 +3378,26 @@ export class EthBancorModule
...
+    reserveAmount: ViewAmount;
+  }): Promise<ProtectionRes> {
+    const depositingNetworkToken = compareString(
+      this.liquidityProtectionSettings.networkToken,
+      reserveAmount.id
     );
 
+    const inputToken = this.token(reserveAmount.id);
+    
+    const { maxStakes } = await this.getMaxStakes({poolId})
```
Comment (Velua) [inline src/store/modules/swap/ethBancor.ts:3399]: @pingustar @H34D is someones prettier config off? ```({poolId})``` should have got auto-formatted to ```({ poolId })```

File: src/components/protection/StakeButtons.vue [hunk 3]
```diff
@@ -32,13 +49,34 @@
...
 
 @Component({
-  components: { GrayBorderBlock, MainButton }
+  components: {LabelContentSplit, GrayBorderBlock, MainButton }
 })
 export default class StakeButtons extends Vue {
   @Prop({ default: false }) showAddLiquidity!: boolean;
 
+  maxStakesBnt: any = 0;
+  maxStakesTkn: any = 0;
```
Comment (Velua) [inline src/components/protection/StakeButtons.vue:64]: Type these

File: src/components/protection/AddProtectionSingle.vue [hunk 4]
```diff
@@ -359,6 +370,29 @@ export default class AddProtectionSingle extends Vue {
...
+    this.loadingMaxStakes = true;
+    try {
+      const result = await vxm.ethBancor.getMaxStakes({
+        poolId: this.pool.id
+      });
+      if (this.token.symbol === "BNT")
+        this.maxStakes = result.maxStakesConverted.maxAllowedBnt;
+      else
+        this.maxStakes =
+          result.maxStakesConverted[`maxAllowedTkn${this.token.symbol}`];
```
Comment (Velua) [inline src/components/protection/AddProtectionSingle.vue:385]: I don't see any advantage in using a dynamic key unless you guys do?

File: src/components/protection/AddProtectionSingle.vue [hunk 5]
```diff
@@ -150,6 +148,10 @@ export default class AddProtectionSingle extends Vue {
     return vxm.bancor.relay(poolId);
   }
 
+  maxStakes: any = null;
```
Comment (Velua) [inline src/components/protection/AddProtectionSingle.vue:151]: Type

**Review by Velua — COMMENTED** [review]
Unless there's any reason for the dynamic key, 

```[`maxAllowedTkn${tknReserve.symbol}`]:``` in ```ethBancor.ts``` it should just be ```maxAllowedTkn```

```maxStakes``` also is passed back to the View Layer but not getting used, unless it has any purpose it shouldn't be there. Especially being in wei form which should be abstracted from the View layer. 

Unless there's a reason for it lets delete ```maxStakes``` from coming back and rename the data of ```maxStakesConverted``` to ```maxStakes```.

**Comment by pingustar** [comment]
https://github.com/bancorprotocol/webapp/commit/a65ee4c66a9b9b42104862aca0a939b29b7ea511
https://github.com/bancorprotocol/webapp/commit/e6926e2588fe6699f6522f4a96bd330fd13485c5

**Pull request merged**